### DISCUSSION TITLE
iOS OpenAL soft with power shell

### DISCUSSION THREAD

**Discussion by qoozta** [axmolengine/axmol#1375] [Q&A] [answered]
In 1.x I can use 
`cmake -S . -B build -GXcode -DCMAKE_TOOLCHAIN_FILE=$AX_ROOT/cmake/ios.toolchain.cmake -DPLATFORM=OS64 -DAX_USE_ALSOFT=ON`
to generate the cpp-tests Xcode project which using OpenAL soft.
Is it possible to use OpenAL soft with power shell script on iOS? How to pass the cmake variable from power shell script?

**Comment by halx99** [comment, marked as answer]
yes, 

`axmol build -p ios -a arm64 -xc "-DAX_USE_ALSOFT=ON" -c`

**Reply by qoozta** [reply]
The Xcode project was generated successfully, however the OpenAL target minimum deployment was set to iOS 11, need to set it to iOS 12 manually, then everything is working properly. Thank you!

**Reply by NgVThangBz** [reply]
@qoozta you can use `XCODE_ATTRIBUTE_IPHONEOS_DEPLOYMENT_TARGET "12"` in Cmake file (set you target in xcode)

**Reply by qoozta** [reply]
Using `axmol build -p ios -a x64 -xc "-DAX_USE_ALSOFT=ON" -c` to generate an iOS Simulator xcode project, then try to compile it, it will shows error in libcommon, but the path is openal/common.
`....../axmol/thirdparty/openal/common/althrd_setname.cpp:43:10 'pthread_np.h' file not found`
The arm64 build was no problem, what is wrong with my settings?
